specs/capri/FileShare_Design.md

```diff
@@ -0,0 +1,369 @@
+
+ “profileId”: “string”
+
+ “poolId”: “string”
+
+ “snapshotId”: “string”
+
+ “exportLocations” : “array”   // path[ ]
+```
+##### 4. PUT /v1beta/{tenantId}/file/shares/{shareId}
+* Updates a share.
+##### Request
+```json
+  “name” : “string”
+
+  “description” : “string”
+```  
+ ##### Response parameters
+ ```json
+ “Id” : “string”
+
+ “protocol” : “string”
+
+ “createdAt” : “2019-03-20T12:49:00.497Z”
+
+ “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+ “tenantId” : “string”
+
+ “userId” : “string“
+
+ “name” : “string”
+
+ “description” : “string”
+
+ “size” : “int64”
+
+ “availabilityZone”: “string”
+
+ “status”: “string”
+
+ “profileId”: “string”
+
+ “poolId”: “string”
+
+ “snapshotId”: “string”
+```
+##### 5. 	DELETE /v1beta/{tenantId}/file/shares/{shareId}
+* Deletes a share
+##### Request
+```json
+    “Id” : “string”
+```
+##### 6.	GET /v1beta/{tenantId}/file/snapshots
+* Lists all share snapshots
+#### Request
+```json
+  “tenantId”: “string”
+```
+ ##### Response parameters
+```json
+ “snapshotId”: “string”
+
+ “protocol” : “string”
+
+ “createdAt” : “2019-03-20T12:49:00.497Z”
+
+ “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+ “name” : “string”
+
+ “description” : “string”
+
+ “shareSize” : ”int64”
+
+ “snapshotSize” : “int64”
+
+ “status”: “string”
+```
+##### 7.	GET /v1beta/{tenantId}/file/snapshots/{snapshotId}
+* Show details of snapshot
+##### Request
+```json
+  “id”: “string”
+```  
+##### Response parameters
+```json
+  “snapshotId”: “string”
+
+  “protocol” : “string”
+
+  “createdAt” : “2019-03-20T12:49:00.497Z”
+
+  “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+  “name” : “string”
+
+  “description” : “string”
+
+  “shareSize” : “int64”
+
+  “snapshotSize” : “int64”
+
+  “status”: “string”
+```
+##### 8.   POST /v1beta/{tenantId}/file/snapshot
+* Creates a snapshot from a share.
+##### Request
+```json
+  “description” : “string”
+
+  “name” : “string”
+
+  “shareId” : “string“
+```
+  ##### Response parameters
+```json  
+  “Id”: “string”
+
+  “shareId” : “string”
+
+  “protocol” : “string”
+
+  “createdAt” : “2019-03-20T12:49:00.497Z”
+
+  “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+  “name” : “string”
+
+  “description” : “string”
+
+  “shareSize” : “int64”
+
+  “snapshotSize” : “int64”
+
+  “status”: “string”
+```
+##### 9.	PUT /v1beta/{tenantId}/file/snapshots/{snapshotId}
+* Updates a share snapshot.
+##### Request
+```json
+  “description” : “string”
+
+  “name” : “string”
+```  
+##### Response parameters
+```json
+  “snapshotId”: “string”
+
+  “shareId” : “string”
+
+  “protocol” : “string”
+
+  “createdAt” : “2019-03-20T12:49:00.497Z”
+
+  “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+  “name” : “string”
+
+  “description” : “string”
+
+  “shareSize” : “int64”
+
+  “snapshotSize” : “int64”
+
+  “status”: “string”
+```
+##### 10.	DELETE /v1beta/{tenantId}/file/snapshots/{snapshotId}
+* Deletes a share
+##### Request
+```json
+  “snapshotId” : “sting”
+```
+##### 11.	POST /v1beta/{tenantId}/file/accesses/
```

Comment:
Suggestion: `accesses` is very easy to mis-spell. Maybe you could try and keep it simpler. Something like `acl`.

Reply:
Hi anvith, after discussion with xing, we decided "accesses". But i like your suggestion. we can have "acls" instead of "accesses". Any more suggestion and feedback are welcome.

Reply:
I like that ideas!

Reply:
I too like the idea Anvith!